File: problems/24/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        dummy = ListNode(-1, head)
        prev = dummy
        
        while head and head.next:
            first, second = head, head.next
            
            prev.next = second
            first.next = second.next
            second.next = first
            
            head = first.next
            prev = first
            
        
        return dummy.next

File: problems/24/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    dummy, prev := &ListNode{-1, nil}, &ListNode{}
    dummy.Next = head
    prev = dummy
    
    for head != nil && head.Next != nil {
        first, second := head, head.Next
        
        first.Next = second.Next
        second.Next = first
        prev.Next = second
        
        prev = first
        head = first.Next
    }
    
    return dummy.Next
}